### PR TITLE
Update version resolution logic

### PR DESCRIPTION
**Pull request by mthalman**
When nearing release, .NET switches to stable branding for its versioning scheme.  This means that the versions of shared frameworks contained in the SDK zip file are not build-specific.  In the case of the upcoming 5.0 release, the versions of runtime and ASP.NET Core being referenced are just labeled as 5.0.0.  This breaks the existing logic in the `get-drop-versions.sh` script because it relies on this value being build-specific in order to generate the correct input to the update-dependencies tool.

I've updated the logic to account for this scenario by not relying on the .version files for the shared frameworks and instead lookup up the build-specific version numbers associated with the SDK build by using the Version.Details.xml file.

This also required changes to the update-dependencies tool because the URL path to the files is different.  The folder name contains the build version but the filename does not.  The simplest way I could think of this was just to use a heuristic that accounts for this URL format in the case when the version string contains `-rtm.`.

### CODE REVIEW

File: eng/get-drop-versions.sh [hunk 1]
```diff
@@ -7,13 +7,23 @@ set -u
 
 channel=$1
 
+sudo apt-get update && \
+    sudo apt-get install -y --no-install-recommends libxml2-utils
+
 curl -SLo sdk.zip https://aka.ms/dotnet/$channel/Sdk/dotnet-sdk-win-x64.zip
 
+commitSha=$(unzip -p sdk.zip "sdk/*/.version" | cat | head -1)
```
Comment (MichaelSimons) [inline eng/get-drop-versions.sh:15]: It's a little inefficient that the script unpacks the sdk `.version` file multiple times.

File: eng/pipelines/update-dependencies.yml [hunk 2]
```diff
@@ -25,9 +25,6 @@ stages:
     - script: >
         docker run --rm -v /var/run/docker.sock:/var/run/docker.sock update-dependencies
         $(dockerfileVersion)
-        --user $(dotnetDockerBot.userName)
```
Comment (MichaelSimons) [inline eng/pipelines/update-dependencies.yml:28]: I'm not understanding why these are being removed.  Can you explain this change?

Comment (mthalman) [reply eng/pipelines/update-dependencies.yml]: That was for testing purposes.  Added this back.

File: eng/get-drop-versions.sh (hunk 1 above)
Comment (mthalman) [reply eng/get-drop-versions.sh]: Fixed

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**